lib/flipper/middleware/memoizer.rb

```diff
@@ -36,11 +37,23 @@ def call(env)
         original = flipper.adapter.memoizing?
         flipper.adapter.memoize = true
 
+        if @opts[:preload_all]
+          names = flipper.features.map(&:name)
+          flipper.preload(names)
+        end
+
+        if @opts[:preload]
+          flipper.preload(@opts[:preload])
+        end
+
         response = @app.call(env)
         response[2] = Rack::BodyProxy.new(response[2]) {
           flipper.adapter.memoize = original
         }
         response
+      rescue
```

Comment:
Ah, I see from bdbee31d I was misreading the `BodyProxy`, thinking it was wrapping the whole `@app.call` not just the body. Makes sense now why it was needed.